src/bson.rs

```diff
@@ -1054,6 +1057,137 @@ impl Binary {
     }
 }
 
+/// Enum of the possible representations to use when converting between Uuid and Binary.
```

Comment:
can we linkify `Uuid` and `Binary` here? You might have to do a regular markdown link for `Uuid`. You can try the shorthand one, and if it doesn't work, `check-rustdoc.sh` will let you know.

Ditto for the other methods. I think its fine to leave the references to "UUID" alone in the docstrings for the enum variants, since those are more generally referring to UUIDs rather than the specific Rust `Uuid` type.

The "link all the things" approach is another one of the Rust API guidelines: https://rust-lang.github.io/api-guidelines/documentation.html#c-link